### PR TITLE
Fix #15096: Crash placing entrance near corner in scenario editor.

### LINKED ISSUE #15096: [3baae4] openrct2.exe: footpath_is_connected_to_map_edge_helper, window_editor_bottom_toolbar_jump_forward_to_invention_list_...
<table>
    <tr>
        <td></td>
        <td><b><a href="https://openrct2.sp.backtrace.io/dashboard/openrct2/project/openrct2/group/3baae43b396c71552f547b61b5ae47a49d33b08dab46071b7690dc31969195b7">New error: [openrct2]</a></b><br /><b>1</b> occurrence since (Fri Nov 15 2019 11:27:39)<br /></td>
    </tr>
</table><table>
    <tr><td><strong>Application</strong></td><td>openrct2.exe</td></tr>
<tr>
    <td><strong>Attributes</strong></td>
    <td>
<strong>error.message</strong> &mdash; 1
        <ul>
                <li><code>EXCEPTION_ACCESS_VIOLATION_READ</code> &mdash; 1</li></ul><br>
<strong>attachment_config.ini</strong> &mdash; 1
        <ul>
                <li><code>C:\Users\Familie de Wit\Documents\OpenRCT2\19c048b6-cbf0-48b8-8de6-9ff4c8955932.ini</code> &mdash; 1</li></ul><br>
<strong>assert_failure</strong> &mdash; 1
        <ul>
                <li><code>*</code> &mdash; 1</li></ul><br>
<strong>commit</strong> &mdash; 1
        <ul>
                <li><code>e0daac9</code> &mdash; 1</li></ul><br>
<strong>fault.address</strong> &mdash; 1
        <ul>
                <li><code>1724614287361</code> &mdash; 1</li></ul><br></td>
    </tr>
    <tr>
        <td><strong>Classifiers</strong></td>
        <td>
            <strong>invalid-read</strong><br></td>
    </tr>
    <tr>
        <td><strong>Frames</strong></td>
        <td>
            <code>
footpath_is_connected_to_map_edge_helper
window_editor_bottom_toolbar_jump_forward_to_invention_list_set_up
widget_invalidate
InputStateWidgetPressed
context_get_width
window_editor_bottom_toolbar_invalidate
GameHandleInputMouse
window_visit_each(std::function&lt;void __cdecl
GameHandleInput
WindowManager::HandleKeyboard
context_handle_input
OpenRCT2::GameState::Update
compute_year
common_gmtime_s&lt;__int64&gt;
common_localtime_s&lt;__int64&gt;
OpenRCT2::Context::Update
InternalFindAtom
OpenRCT2::Context::RunGameLoop
OpenRCT2::Context::RunOpenRCT2
NormalisedMain
wmain</code>
        </td>
    </tr></table>

### PR DESCRIPTION
**Pull request by ZehMatt**
I was unable to reproduce the issue there is probably a specific setup to go out of bounds, the logic was wrong anyway. I think this issue has been there for a while.

### CODE REVIEW

File: src/openrct2/world/Footpath.cpp [hunk 1]
```diff
@@ -1426,8 +1426,7 @@ static int32_t footpath_is_connected_to_map_edge_helper(CoordsXYZ footpathPos, i
                         continue;
                     }
                     edges &= tileElement[elementIndex].AsBanner()->GetAllowedEdges();
-                    ++elementIndex;
-                } while (!tileElement[elementIndex].IsLastForTile());
+                } while (!tileElement[elementIndex++].IsLastForTile());
```
Comment (duncanspumpkin) [inline src/openrct2/world/Footpath.cpp:1429]: L1425 also needs deleted.

File: src/openrct2/world/Footpath.cpp [hunk 2]
```diff
@@ -1426,8 +1426,7 @@ static int32_t footpath_is_connected_to_map_edge_helper(CoordsXYZ footpathPos, i
                         continue;
```
Comment (duncanspumpkin) [inline src/openrct2/world/Footpath.cpp:1426]: L1417 isn't checking if the tileElement is the last element either. There are a few odd/wrong things with this.

Comment (ZehMatt) [reply src/openrct2/world/Footpath.cpp:1426]: Hm right it should definitely start at the first tile, it probably started at 1 to always skip the surface.

Comment (duncanspumpkin) [reply src/openrct2/world/Footpath.cpp:1425]: No sorry it's meant to start at 1 it's just it's meant to check 0 to make sure it's not the last. At this point in the function we already point at a path. What this loop is meant to do is find all the banners on that path.

**Review by duncanspumpkin — APPROVED** [review]

File: distribution/changelog.txt [hunk 3]
```diff
@@ -1,5 +1,6 @@
 0.3.4.1+ (in development)
 ------------------------------------------------------------------------
+- Fix: [#15096] Crash placing entrances in the scenario editor near the map corner.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:3]:
```suggestion
- Fix: [#15096] Crash when placing entrances in the scenario editor near the map corner.
```

**Pull request merged**